content/en/references/lambda-provider-v2.md

```diff
@@ -0,0 +1,223 @@
+It was primarily used as a fallback if the Docker socket was unavailable in the LocalStack container.
+Hence, many users unintentionally used the local executor mode despite configuring `LAMBDA_EXECUTOR=docker`.
+The new provider requires no such configuration and now behaves like the old `docker-reuse` executor.
+
+**No container restart after each invocation:**
+Lambda containers are reused between invocations.
+Therefore, filesystem changes (for example, in `/tmp`) persist between subsequent invocations if dispatched to the same container.
+It is called a "warm start" (see [Operating Lambda](https://aws.amazon.com/blogs/compute/operating-lambda-performance-optimization-part-1/)).
+If you want to force "cold starts", you could set `LAMBDA_KEEPALIVE_MS` to 0 milliseconds.
+
+**Official AWS-provided Lambda images:**
+The new provider uses the official AWS [base images for Lambda](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-images.html) pulled from `public.ecr.aws/lambda/` instead of `lambci` images.
+Hence, Lambda functions' filesystem now matches the AWS Lambda production environment.
+
+**ARM64 lambda functions:**
+Lambda now supports ARM containers for compatible runtimes based on Amazon Linux 2.
+ARM-compatible hosts can create functions with the `arm64` architecture.
+More details are provided under [ARM64 support]({{< ref "references/arm64-support" >}}).
+
+**Transparent endpoint injection in LocalStack Pro:**
+Lambda functions resolve AWS domains such as `s3.amazonaws.com` to the LocalStack container.
+This domain resolution is now completely DNS-based and can be disabled by setting `DNS_ADDRESS=0`.
+For more details, please refer to [Transparent Endpoint Injection]({{< ref "user-guide/tools/transparent-endpoint-injection" >}}).
+The old provider provided patched AWS SDKs to redirect AWS API transparently calls to LocalStack.
+
+## Configuration
+
+The new provider offers new configuration options documented under [Configuration]({{< ref "references/configuration#lambda" >}}).
+Please open a [feature request on GitHub](https://github.com/localstack/localstack/issues/new/choose) and motivate your scenario if you miss something.
+
+The following configuration options from the old provider are discontinued in the new provider:
+* `LAMBDA_EXECUTOR` and, in particular, `LAMBDA_EXECUTOR=local` should be removed (see "Local executor mode is discontinued" above).
+* `LAMBDA_STAY_OPEN_MODE` is now the default behavior and can be removed. Instead, `LAMBDA_KEEPALIVE_MS` can be used to configure how long containers should be kept running in-between invocations.  
+* `LAMBDA_REMOTE_DOCKER` is not used anymore because the new provider always copies zip files and automatically configures hot reloading.
+* `LAMBDA_CODE_EXTRACT_TIME` is no longer used because function creation now happens asynchronously.
+* `LAMBDA_DOCKER_DNS` is currently not supported.
```

Comment:
Guess we could integrate this into the docker flags at some point. It will break transparent endpoint injection though

Reply:
yeah, adding on-demand seems reasonable. Related issue: https://github.com/localstack/localstack/issues/7539